src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -264,7 +268,7 @@ void Battleground::Update(uint32 diff)
             else
             {
                 _ProcessResurrect(diff);
-                if (sBattlegroundMgr->GetPrematureFinishTime() && (GetPlayersCountByTeam(TEAM_ALLIANCE) < GetMinPlayersPerTeam() || GetPlayersCountByTeam(TEAM_HORDE) < GetMinPlayersPerTeam()))
```

Comment:
```
if (sWorld->getBoolConfig(BATTLEGROUND_CROSSFACTION_ENABLED)
{
    if (sBattlegroundMgr->GetPrematureFinishTime() && (GetAlliancePlayersCount() == 0 || GetHordePlayersCount() == 0)))
        _ProcessProgress(diff);
}
else
{
    if (sBattlegroundMgr->GetPrematureFinishTime() && (GetPlayersCountByTeam(TEAM_ALLIANCE) < GetMinPlayersPerTeam() || GetPlayersCountByTeam(TEAM_HORDE) < GetMinPlayersPerTeam()))
        _ProcessProgress(diff);
}
```
or better

```
#ifdef CFBG
if (sBattlegroundMgr->GetPrematureFinishTime() && (GetAlliancePlayersCount() == 0 || GetHordePlayersCount() == 0))
#else
if (sBattlegroundMgr->GetPrematureFinishTime() && (GetPlayersCountByTeam(TEAM_ALLIANCE) < GetMinPlayersPerTeam() || GetPlayersCountByTeam(TEAM_HORDE) < GetMinPlayersPerTeam()))
#endif
```

Reply:
end etc over pr, no need change logic of queue when CFBG not enable,

Reply:
Thanks for your feedback